presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleTTLNodeSelector.java

```diff
@@ -0,0 +1,284 @@
+        NodeMap nodeMap = this.nodeMap.get().get();
+        NodeAssignmentStats assignmentStats = new NodeAssignmentStats(nodeTaskMap, nodeMap, existingTasks);
+
+        List<InternalNode> eligibleNodes = getEligibleNodes(maxTasksPerStage, nodeMap, existingTasks);
+        NodeSelection randomNodeSelection = new RandomNodeSelection(eligibleNodes, minCandidates);
+
+        boolean splitWaitingForAnyNode = false;
+
+        OptionalInt preferredNodeCount = OptionalInt.empty();
+        for (Split split : splits) {
+            if (split.getNodeSelectionStrategy() != NodeSelectionStrategy.NO_PREFERENCE) {
+                throw new PrestoException(
+                        NODE_SELECTION_NOT_SUPPORTED,
+                        format("Unsupported node selection strategy for TTL scheduling %s", split.getNodeSelectionStrategy()));
+            }
+
+            List<InternalNode> candidateNodes = randomNodeSelection.pickNodes(split);
+            if (candidateNodes.isEmpty()) {
+                log.warn("No nodes available to schedule %s. Available nodes %s", split, nodeMap.getActiveNodes());
+                throw new PrestoException(NO_NODES_AVAILABLE, "No nodes available to run query");
+            }
+
+            Optional<InternalNodeInfo> chosenNodeInfo = simpleNodeSelector.chooseLeastBusyNode(
+                    candidateNodes,
+                    assignmentStats::getTotalSplitCount,
+                    preferredNodeCount,
+                    maxSplitsPerNode,
+                    assignmentStats);
+            if (!chosenNodeInfo.isPresent()) {
+                chosenNodeInfo = simpleNodeSelector.chooseLeastBusyNode(
+                        candidateNodes, assignmentStats::getQueuedSplitCountForStage, preferredNodeCount, maxPendingSplitsPerTask, assignmentStats);
+            }
+
+            if (chosenNodeInfo.isPresent()) {
+                split = new Split(
+                        split.getConnectorId(),
+                        split.getTransactionHandle(),
+                        split.getConnectorSplit(),
+                        split.getLifespan(),
+                        new SplitContext(chosenNodeInfo.get().isCacheable()));
+
+                InternalNode chosenNode = chosenNodeInfo.get().getInternalNode();
+                assignment.put(chosenNode, split);
+                assignmentStats.addAssignedSplit(chosenNode);
+            }
+            else {
+                splitWaitingForAnyNode = true;
+            }
+        }
+
+        ListenableFuture<?> blocked = splitWaitingForAnyNode ?
+                toWhenHasSplitQueueSpaceFuture(existingTasks, calculateLowWatermark(maxPendingSplitsPerTask)) : immediateFuture(null);
+
+        return new SplitPlacementResult(blocked, assignment.build());
+    }
+
+    @Override
+    public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks, BucketNodeMap bucketNodeMap)
```

Comment:
Seems like this means queries with bucketed execution don't utilize TTLs.  Is that correct?